attestation-agent/deps/resource_uri/src/lib.rs

```diff
@@ -160,6 +167,23 @@ mod tests {
             repository: "alice".into(),
             r#type: "cosign-key".into(),
             tag: "213".into(),
+            query: None,
+        };
+        assert_eq!(expected, resource);
+    }
+
+    const TEST_QS_URL: &str = "kbs:///plugin/plugname/resourcename?param1=value1&param2=value2";
+
+    #[test]
+    fn deserialize_qs() {
```

Comment:
rstest could help to make the tests more dry. I am sure that you will love it : )

examples https://github.com/confidential-containers/guest-components/blob/main/attestation-agent/attestation-agent/src/eventlog.rs#L158-L174

Reply:
thanks, rstest definitely makes the code cleaner. Let me rework that test  case